packages/core/src/logs/storage.ts

```diff
@@ -183,22 +186,97 @@ export async function queryLogs(
 }
 
 /**
- * Get server aggregations
+ * Get server aggregations with status
+ *
+ * @param db - Database instance
+ * @param registryServers - Optional list of registered server names
+ * @returns Server information with status (online/offline/not-found)
  */
 export async function getServers(
   db: BunSQLiteDatabase<typeof schema>,
+  registryServers?: string[],
 ): Promise<ServerInfo[]> {
-  const result = await db
+  // Get servers that have logs in the database, with health data
+  const logsResult = await db
     .select({
       name: logs.serverName,
       logCount: count(logs.id),
       sessionCount: sql<number>`COUNT(DISTINCT ${logs.sessionId})`,
+      health: serverHealth.health,
     })
     .from(logs)
-    .groupBy(logs.serverName)
-    .orderBy(logs.serverName);
+    .leftJoin(serverHealth, eq(logs.serverName, serverHealth.serverName))
+    .groupBy(logs.serverName, serverHealth.health)
+    .orderBy(sql`LOWER(${logs.serverName}) COLLATE NOCASE`);
+
+  // Create a map of registry servers (normalized name -> original name)
+  // This preserves the registry's casing as the source of truth
+  const registryMap = new Map<string, string>();
+  const registryProvided = Array.isArray(registryServers);
+  if (registryProvided) {
+    for (const serverName of registryServers || []) {
+      registryMap.set(serverName.toLowerCase(), serverName);
+    }
+  }
+
+  // Create a map of servers from logs
+  const serverMap = new Map<string, ServerInfo>();
+
+  // Add servers from logs with their counts
+  for (const server of logsResult) {
+    const normalizedName = server.name.toLowerCase();
+    const registryName = registryMap.get(normalizedName);
+
+    // Determine status based on registry membership and health
+    let status: "online" | "offline" | "not-found";
+    if (registryProvided && registryName) {
+      // Server is in registry - check health to determine online vs offline
+      const health = server.health as HealthStatus | null;
```

Comment:
see my comment about using `text({ enum: ["up", "down"] })` for this column - could be helpful